src/dlgRoomProperties.cpp

```diff
@@ -0,0 +1,555 @@
+    // Parse the initial text before the curly braces containing count
+    QRegularExpression countStripper(qsl("^(.*) {.*}$"));
+    QRegularExpressionMatch match = countStripper.match(newSymbolText);
+    if (match.hasMatch() && match.lastCapturedIndex() > 0) {
+        return match.captured(1);
+    }
+    return newSymbolText;
+}
+
+
+int dlgRoomProperties::getNewWeight()
+{
+    if (mpWeights.size() <= 1) {
+        return spinBox_weight->value();
+    }
+    QString newWeightText = comboBox_weight->currentText();
+    if (newWeightText == multipleValuesPlaceholder) {
+        return -1; // User did not want to select any weight, so we will do no change
+    }
+    // Parse an initial number out of what was selected or typed
+    QRegularExpression countStripper(qsl("^\\s*(\\d+)"));
+    QRegularExpressionMatch match = countStripper.match(newWeightText);
+    if (match.hasMatch() && match.lastCapturedIndex() > 0) {
+        return match.captured(1).toInt();
+    }
+    if (newWeightText.toInt() > 0) {
+        return newWeightText.toInt();
+    }
+    return -2; // Maybe some other input we did not understand, so we will do no change
+}
+
+
+void dlgRoomProperties::slot_updatePreview()
+{
+    auto realSymbolColor = selectedSymbolColor != nullptr ? selectedSymbolColor : defaultSymbolColor();
+    QString newSymbol = getNewSymbol();
+    if (newSymbol == multipleValuesPlaceholder) {
+        newSymbol = QString();
+        pushButton_setSymbolColor->setStyleSheet(QString());
+    } else {
+        pushButton_setSymbolColor->setStyleSheet(
+        qsl("background-color: %1; color: %2; border: 1px solid; border-radius: 1px;")
+            .arg(realSymbolColor.name(), backgroundBasedColor(realSymbolColor).name()));
+    }
+    label_preview->setFont(getFontForPreview(newSymbol));
+    label_preview->setText(newSymbol);
+    label_preview->setStyleSheet(
+        qsl("color: %1; background-color: %2; border: %3;")
+            .arg(realSymbolColor.name(), mRoomColor.name(), mpHost->mMapperShowRoomBorders ? qsl("1px solid %1").arg(mpHost->mRoomBorderColor.name()) : qsl("none")));
+
+}
+
+
+QFont dlgRoomProperties::getFontForPreview(QString symbolString)
+{
+    auto font = mpHost->mpMap->mMapSymbolFont;
+    font.setPointSize(font.pointSize() * 0.9);
+    if (!symbolString.isEmpty()) {
+        QFontMetrics mapSymbolFontMetrics = QFontMetrics(font);
+        QVector<quint32> codePoints = symbolString.toUcs4();
+        QVector<bool> isUsable;
+        for (int i = 0; i < codePoints.size(); ++i) {
+            isUsable.append(mapSymbolFontMetrics.inFontUcs4(codePoints.at(i)));
+        }
+        bool needToFallback = isUsable.contains(false);
+        if (needToFallback) {
+            symbolString = QString(QChar::ReplacementCharacter);
+            font.setStyleStrategy(static_cast<QFont::StyleStrategy>(mpHost->mpMap->mMapSymbolFont.styleStrategy() & ~(QFont::NoFontMerging)));
+        }
```

Comment:
In this situation `symbolString` is not a reference so modifying it is not going to have any effect - so the line is dead code and what is left can be simplified, at least, to:
```suggestion
        if (isUsable.contains(false)) {
            font.setStyleStrategy(static_cast<QFont::StyleStrategy>(mpHost->mpMap->mMapSymbolFont.styleStrategy() & ~(QFont::NoFontMerging)));
        }
```

Reply:
Looking forward to second opinions here!
I can't say I grok references, and this is not even my code, but simply copied as-is from `dlgRoomSymbol::getFontForPreview(QString text)` (which will soon be removed) so your criticism probably applies there, too? (Please don't rush to fix it there as that will soon be removed anyway).

Reply:
Well it was factored out to a separate method I think by me or Delwing when we created the separate `dlgRoomSymbol` class or enabled the text to be coloured (respectively) - and it looks like that error is mine. So yeah, discard my suggestion for now...